dashboard/lib/services/lti/account_linker.rb

```diff
@@ -0,0 +1,27 @@
+module Services
+  module Lti
+    class AccountLinker < Services::Base
+      attr_reader :user, :session
+
+      def initialize(user:, session:)
+        @user = user
+        @session = session
+      end
+
+      def call
+        ActiveRecord::Base.transaction do
+          user.authentication_options << rehydrated_user.authentication_options.first
+          Services::Lti.create_lti_user_identity(user)
+          if user.user_type == ::User::TYPE_TEACHER
+            user.lti_roster_sync_enabled = true
+          end
```

Comment:
https://github.com/code-dot-org/code-dot-org/blob/b6e1a2bee8f770e52c337b5a8ae0660f03ee5ab0/dashboard/app/models/user.rb#L1498-L1500
```suggestion
          user.lti_roster_sync_enabled = true if user.teacher?
```